clients/src/main/java/org/apache/kafka/clients/consumer/CooperativeStickyAssignor.java

```diff
@@ -53,9 +56,34 @@ public String name() {
         return Arrays.asList(RebalanceProtocol.COOPERATIVE, RebalanceProtocol.EAGER);
     }
 
+    @Override
+    public void onAssignment(Assignment assignment, ConsumerGroupMetadata metadata) {
+        this.generation = metadata.generationId();
+    }
+
+    @Override
+    public ByteBuffer subscriptionUserData(Set<String> topics) {
+        if (generation == DEFAULT_GENERATION) {
```

Comment:
Should we create a schema class, not direct ByteBuffer for adding a field easily in the future?

Reply:
Yeahh...will do